### PR TITLE
[SPARK-46312][CORE] Use `lower_camel_case` in `store_types.proto`

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to use 'lower_camel_case` in `store_types.proto`.

### Why are the changes needed?

According to our guideline, we had better follow [FIELD_LOWER_SNAKE_CASE](https://buf.build/docs/lint/rules#field_lower_snake_case)

https://github.com/apache/spark/blob/9585cf6d56e3af37142609668dda1eeda3ec876f/core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto#L23

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @LuciferYang and @gengliangwang

File: core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto [hunk 1]
```diff
@@ -164,7 +164,7 @@ message ExecutorStageSummaryWrapper {
 message ExecutorResourceRequest {
   optional string resource_name = 1;
   int64 amount = 2;
-  optional string discoveryScript = 3;
+  optional string discovery_script = 3;
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:167]: cc @hvanhovell @grundprinzip

[the remark this answers is not in the document]
Comment (dongjoon-hyun) [reply core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:167]: No~ ProtocolBuffer is safe with the field renaming, @yaooqinn .

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon and @yaooqinn .
Merged to master.

File: core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (yaooqinn) [reply core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:167]: thanks for explanation:)

**Comment by LuciferYang** [comment]
late LGTM, thanks @dongjoon-hyun ~

**Comment by gengliangwang** [comment]
Late LGTM!

**Pull request closed without merging**